crates/p2p/src/swarm/errors.rs

```diff
@@ -23,26 +23,37 @@ pub enum Error {
 /// Validation errors.
 #[derive(Debug, Error)]
 pub enum ValidationError {
+    /// The signature is invalid.
     #[error("Invalid signature")]
     InvalidSignature,
+    /// The message signer is not in the signer's allowlist.
     #[error("Not in signers allowlist")]
     NotInSignersAllowlist,
```

Comment:
You're removing `signer_allow_list` in `crates/p2p/src/swarm/mod.rs` on L114 in this PR. Does it make sense to keep this error variant here?

If yes, note that this whole error will have a single variant, `InvalidSignature`, thus does it makes sense to keep it as a `enum` with that single variant and not convert it to a "singleton" error?